lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,227 @@
+        child: Padding(
+          padding: const EdgeInsets.all(8),
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: [
+              Center(
+                child: Text('Settings',
+                    style: Theme.of(context).textTheme.headlineMedium),
+              ),
+              // Add a dropdown button for setting a new reference scale.
+              DropdownButton(
+                alignment: Alignment.center,
+                hint: const Text(
+                  'Select a New Reference Scale',
+                  style: TextStyle(
+                    color: Colors.deepPurple,
+                  ),
+                ),
+                // Set the selected scale
+                value: _selectedScale,
+                icon: const Icon(
+                  Icons.arrow_drop_down,
+                  color: Colors.deepPurple,
+                ),
+                elevation: 16,
+                style: const TextStyle(color: Colors.deepPurple),
+                // Set the onChanged callback to update the selected scale.
+                onChanged: (newScale) {
+                  updateScale(newScale!.toDouble());
+                },
+                items: _referenceScaleList,
+              ),
+              const Text(
+                'Apply Reference Scale to a Layer',
+              ),
+              // Add a list of checkboxes for selecting feature layers that will honor the reference scale.
+              Column(
+                children: [
+                  // Create a checkbox for each feature layer.
+                  for (var layer in _allFeatureLayers)
+                    CheckboxListTile(
+                      value: _selectedFeatureLayers.contains(layer),
+                      onChanged: (value) {
+                        // Update the selected feature layers.
+                        setNewState(() {
+                          if (value ?? false) {
+                            _selectedFeatureLayers.add(layer);
+                          } else {
+                            _selectedFeatureLayers.remove(layer);
+                          }
+                        });
+                      },
+                      title: Text(layer),
+                    ),
+                ],
+              ),
+              // Add a button to set the map scale to the reference scale and update the layers.
+              ElevatedButton(
+                onPressed: () {
+                  setScaleAndLayers();
+                  Navigator.pop(newContext);
+                },
+                child: const Text('Set Map Scale to Reference Scale'),
+              ),
+            ],
+          ),
+        ),
+      );
+    });
+  }
+
+  void updateScale(double newScale) {
+    // Update the scale.
+    setState(() => _scale = newScale);
+  }
+
+  void setScaleAndLayers() {
+    // Set the map scale to the reference scale and update the layers.
+    setState(() {
```

Comment:
I just tested this and you are right! But thought since ArcGISMapView() is inside the widget tree, we would need to redraw it to update the scale and the layers?

Reply:
We do our stuff via the _mapViewController, which kind of bypasses the setState/build mechanism.

Flutter does the same thing with some of its widgets, like the TextEditingController. Like you can call `_textEditingController.clear()` without having to call `setState()`. We adopted that idea for our _mapViewController. There's just so much stuff going on in a _mapViewController that we wouldn't want to build that all in to the widget tree.

Reply:
That makes sense, thanks for the explanation!